File: backend/api/class.py
from fastapi import APIRouter
from typing import List

from crud.classes import ClassCRUD
import schemas.classes as class_schema

router = APIRouter(prefix="/classes", tags=["classes"])

@router.get("", response_model=List[class_schema.Base])
async def get_classes(db: ClassCRUD):
    return await db.get_classes()

@router.get("/{classId}", response=class_schema.Base)
async def get_class_by_id(classId: int, db: ClassCRUD):
    return await db.get_class_by_id(classId)

@router.post("")
async def create_class(new_class: class_schema.Create, db: ClassCRUD):
    return await db.create_class(new_class)

@router.put("")
async def update_class(classId: int, class_: class_schema.Update, db: ClassCRUD):
    return await db.update_class(classId, class_)

@router.delete("")
async def delete_class(classId: int, db: ClassCRUD):
    return await db.delete_class(classId)

File: frontend/src/api/class.js
import request from './req';
import axios from 'axios';

export const getClasses = (token) => request('GET', `/classes?token=${token}`);
export const createClass = (data, token) => request('POST', `/classes?token=${token}`, data);
export const updateClass = (classId, data, token) => request("PUT", `/classes/${classId}?token=${token}`, data);
export const deleteClass = (id, token) => request("DELETE", `/classes/${id}?token=${token}`);